index.js

```diff
@@ -72,12 +72,22 @@ export default function baconRouter(baseUrl, initialPath, ...routesAndReturns) {
     return history.flatMapLatest((history) => {
         let {location/*, state*/} = history;  // eslint-disable-line spaced-comment
         const currentRoute = location.replace(baseUrl, ''); // @TODO Less hacky.
+        const [, encodedPath = '', search = '', hash = ''] = /^([^?#]*)(?:\?([^#]*))?(?:#(.*))?$/.exec(currentRoute);
+
+        let path;
+
+        try {
+            path = decodeURIComponent(encodedPath);
+        } catch (error) {
+            // URL path isn't valid
+            return bacon.never();
```

Comment:
😓 will this now cause invalid routes not to resolve/render at all?

Reply:
Hm yeah, it should probably be a `bacon.Error` so we can choose how to handle it.